src/main/java/net/minecraftforge/event/world/BlockEvent.java

```diff
@@ -282,7 +283,38 @@ public CreateFluidSourceEvent(World world, BlockPos pos, IBlockState state)
             super(world, pos, state);
         }
     }
-    
+
+    /**
+     * Fired when a liquid places a block.
```

Comment:
This needs documentation for which cases this fires from vanilla. While I would have expected obsidian, cobble and stone, the fire surprised me.

Reply:
Also, it would be good to clarify in the comments how to "cancel" the event. Since it isn't actually cancelable and also doesn't use a Result return value, I guess if you want to prevent the placement then you return the actual state of the block that is already there, correct?